internal/main.go

```diff
@@ -47,6 +49,7 @@ func main() {
 	flag.StringVar(&flags.root, "root", "/", "root of the filesystem")
 	flag.Var(&flags.stage, "stage", fmt.Sprintf("execution stage. %v", stages.Names()))
 	flag.BoolVar(&flags.version, "version", false, "print the version and exit")
+	flag.StringVar(&flags.validate, "validate", "", "path to ignition config to validate")
```

Comment:
should a note be added along the lines of "for development use"?

Reply:
If a user wanted a command line way to validate configs, this wouldn't be a bad way for them to go about it

Reply:
Except if you run it, you get a whole bunch of errors (it the fact it can't open the config cache) which is confusing and not a good experience.

Reply:
If this approach is still running too much of Ignition to be maintainable, maybe the noop provider idea wasn't a good one and we should have a completely different code path for this.  If this mechanism is the only way to validate an Ignition config from the command line, people will use it whether it's for "development" or not.

Reply:
The engine code has the advantage of fetching and merging referenced config, which could be useful. If we wrote it into a seperate code path, we would either lose that or duplicate a bunch of code. On the other hand, that's not something we've needed to test (recently).

Reply:
There are blackbox tests that check that.

I assume https://coreos.com/validate/ doesn't follow references?

More generally, if someone wants to validate an Ignition config, is it necessarily true that they want to follow references?  Might there be endpoints which are accessible from the target machine but not their dev machine?

Reply:
> I assume https://coreos.com/validate/ doesn't follow references?

Correct.

> More generally, if someone wants to validate an Ignition config, is it necessarily true that they want to follow references? Might there be endpoints which are accessible from the target machine but not their dev machine?

I was referring more to people developing Ignition wanting to test using it, but again, I'm not sure that's much of an issue.

I'll open a new PR with it as its own code path, not fetching external configs.